### ISSUE TITLE
`ProxyManager::getProxy` adds elements to the map if the input `ID` doesn't already exist.

### ISSUE THREAD

**Issue by sgilmore10** [mathworks/libmexclass#62]
`ProxyManager::getProxy` adds elements to the map if the input `ID` doesn't already exist. Here's how this method is currently implemented:

```cpp
std::shared_ptr<Proxy> ProxyManager::getProxy(ID id) {
    return ProxyManager::singleton.proxy_map[id];
}
```

 The problem is that `operator[]` will default insert key-value pairs into the map if the key does not already exist. See the example in the[ cpp documentation](https://en.cppreference.com/w/cpp/container/unordered_map/operator_at).  We should be using `find()`to lookup proxies in the map. 

Here's my proposed implementation:

```cpp
std::shared_ptr<Proxy> ProxyManager::getProxy(ID id) const {
    auto itr = proxyManager::singleton.proxy_map.find(id);
    if (itr->first != proxyManager::singleton.proxy_map.end()) {
        return itr->second;
    } else {
        return nullptr;
    }
}
```

We could further enhance this method to return `std::variant<std::shared_ptr<Proxy>, libmexclass::error::Error>` instead of a `std::shared_ptr<Proxy>`.

**Comment by sgilmore10** [comment]
Actually, I forgot that `getProxy` is a `static` method and therefore cannot be `const`. However, we should still use `find` instead of `operator[]`.

**Comment by sgilmore10** [comment]
take

**Comment by kevingurney** [comment]
Closed by #63.
